### ISSUE TITLE
Signal to reopen log files (logrotate)

### ISSUE THREAD

**Issue by dvershinin** [xmrig/xmrig-proxy#370]
xmrig-proxy is configured with:

    ...
    "access-log-file": "/var/log/xmrig-proxy/access.log",
    ...

There is logrotate in place to rotate this daily, e.g. in `/etc/logrotate.d/xmrig-proxy`:

```
/var/log/xmrig-proxy/*.log {
        daily
        missingok
        rotate 52
        compress
        delaycompress
        notifempty
        create 640 xmrig-proxy adm
        sharedscripts
}
```

However, sure enough, xmrig-proxy does not see when files are being rotated, so e.g.
when logrotate moves `/var/log/xmrig-proxy/access.log` to `/var/log/xmrig-proxy/access.log-20191214`, then xmrig-proxy keeps logging to the `/var/log/xmrig-proxy/access.log-20191214` because it is just using the old filehandle.

It is common for a program to support some signal in order to only reopen log files, e.g. `-USR1` or `-HUP`. 

However, I tried using those, e.g. `kill -USR1 <xmrig-proxy-PID>` or `kill -HUP <xmrig-proxy-PID>`, but this results in `xmrig-proxy` complete restart, which is undesired.

Is it possible to have xmrig-proxy only reopen log files? I think this is useful for config re-read as well. (I see there is [config watch](https://github.com/xmrig/xmrig-proxy/issues/119) but the standard approach is sending a process signal for "reloading" files used by an application.

Expected behavior after running `kill -HUP <xmrig-proxy-PID>`:

* PID stays the same (the process is not restarted)
* xmrig-proxy reopens the log file (new handle), thus logging to the same file name as it is configured ( `/var/log/xmrig-proxy/access.log` )
* reloads config file (useful even when `watch` is `false` in config)

**Comment by xmrig** [comment]
Good idea, I am currently busy with the miner and big proxy update delayed, I will implement it.
Thank you.
